### PR TITLE
HW05

### PR DESCRIPTION
**Pull request by DanielHalachev**
Added source code and screenshots for HW05.

### CODE REVIEW

File: Images/HW05/scene1.txt
Comment (VladislavVulchevChaos) [inline Images/HW05/scene1.txt:1]: Very nice to use a scene file

File: SourceCode/include/tracer/Ray.h [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#include "tracer/Triangle.h"
+
+struct Ray {
+  constexpr static const float FLOAT_EPSILON = 1e-8;
+  Vector origin;
+  Vector direction;
+
+  Ray();
+  Ray(const Vector &origin, const Vector &direction);
+  std::optional<Vector> intersectWithTriangle(const Triangle &triangle) const;
```
Comment (VladislavVulchevChaos) [inline SourceCode/include/tracer/Ray.h:16]: Usually it's better for this function to live in the Primitive that is intersected.

File: SourceCode/src/RayTracer.cpp [hunk 2]
```diff
@@ -32,13 +40,31 @@ void RayTracer::generateRays() {
...
                << "\n"
-               << this->image.width << " " << this->image.height << "\n";
+               << this->image.width << " " << this->image.height << "\n"
+               << 255 << "\n";
   for (unsigned int pixelRow = 0; pixelRow < image.height; pixelRow++) {
     for (unsigned int pixelCol = 0; pixelCol < image.width; pixelCol++) {
-      outputStream << this->pixelRays[pixelRow][pixelCol].generateColor() << "\t";
+      std::vector<Vector> points;
+      for (auto &triangle : this->sceneTriangles) {
+        std::optional<Vector> intersectionPoint = this->pixelRays[pixelRow][pixelCol].intersectWithTriangle(triangle);
```
Comment (VladislavVulchevChaos) [inline SourceCode/src/RayTracer.cpp:52]: Decouple the intersection and the generation of the final image.

File: SourceCode/include/tracer/Ray.h (hunk 1 above)
Comment (DanielHalachev) [reply SourceCode/include/tracer/Ray.h:16]: Will the reason why be covered in future lectures of the course? If not, could you briefly explain?

**Pull request merged**